db/openshift.deploy.yml

```diff
@@ -326,3 +329,86 @@ objects:
               dnsPolicy: "ClusterFirst"
               serviceAccountName: "${JOB_SERVICE_ACCOUNT}"
               serviceAccount: "${JOB_SERVICE_ACCOUNT}"
+  - kind: CronJob
+    apiVersion: "batch/v1"
+    metadata:
+      name: ${NAME}-${ZONE}-${COMPONENT}-restore
+      labels:
+        app: ${NAME}-${ZONE}
+        cronjob: ${NAME}-${ZONE}
+    spec:
+      schedule: "0 0 31 2 *" # At 00:00 on day-of-month 31 in February, so it won't run, we'll trigger it manually when need
+      concurrencyPolicy: "Replace"
+      successfulJobsHistoryLimit: ${{SUCCESS_JOBS_HISTORY_LIMIT}}
+      failedJobsHistoryLimit: ${{FAILED_JOBS_HISTORY_LIMIT}}
+      jobTemplate:
+        metadata:
+          labels:
+            app: ${NAME}-${ZONE}
+            cronjob: ${NAME}-${ZONE}
+        spec:
+          backoffLimit: ${{JOB_BACKOFF_LIMIT}}
+          template:
+            metadata:
+              labels:
+                app: ${NAME}-${ZONE}
+                cronjob: ${NAME}-${ZONE}
+            spec:
+              containers:
+                - name: ${NAME}-${ZONE}-${COMPONENT}-restore
+                  # use the same image as our database, so we can run the psql command
+                  image: image-registry.apps.silver.devops.gov.bc.ca/${OC_NAMESPACE}/${NAME}-${ZONE}-${COMPONENT}:${ZONE}-db
+                  command: ["/bin/sh", "-c"]
+                  args:
+                  - |
+                    sql_zip_file=$(find backups/${RESTORE_DIR} -type f -name "*.sql.gz" -print0)
+                    echo "Found zipped backup file: $sql_zip_file"
+                    gzip -d $sql_zip_file
+                    sql_file=$(find backups/${RESTORE_DIR} -type f -name "*.sql" -print0)
+                    echo "Running SQL file: $sql_file"
+                    psql -h ${TARGET_HOST} -U ${POSTGRES_USER} -d ${POSTGRES_DB} -f $sql_file
+                    gzip $sql_file
+                    echo "Finish database restore"
+                  volumeMounts:
+                    - name: ${NAME}-${ZONE}-${COMPONENT}-restore
+                      mountPath: "${BACKUP_DIR}"
+                    - name: ${NAME}-${ZONE}-${COMPONENT}-restore
+                      mountPath: /var/lib/postgresql
```

Comment:
Maybe only because I am not familiar so get confused... why are two 'mountPath's on the same volume "${NAME}-${ZONE}-${COMPONENT}-restore"?

Reply:
I got the same confusion before, but I can only give the same name, whatever other names I tried, it will return me an error. So I think it's because the volume for this container is named `${NAME}-${ZONE}-${COMPONENT}-restore`, and we need to mount both backup directory to get all the backup files, and mount the Postgres directory to get the Postgres command line tool
<img width="553" alt="Screen Shot 2024-08-20 at 10 16 40 AM" src="https://github.com/user-attachments/assets/6d9405a0-7ce3-41db-9014-2dff92ce97c6">

Reply:
removed the PostgreSQL mount as no need